packages/core/src/root.ts

```diff
@@ -697,8 +698,6 @@ function makeValueHotSwappable(
     return wrappedFunction;
   }
 
-  seen.set(value, value);
```

Comment:
This was setting the values twice in the map, once with the wrapped value and once with the initial value. Removing this didn't make any tests fail. If it was supporting a real case, I can add tests to make sure its covered.